tools-local/tasks/core-setup.tasks.csproj

```diff
@@ -15,16 +15,17 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="NuGet.ProjectModel" Version="$(NugetProjectModelPackageVersion)" />
+    <PackageReference Include="NuGet.ProjectModel" Version="4.9.4" />
```

Comment:
These are pinned as far as Maestro++/Darc are concerned: https://github.com/dotnet/core-setup/blob/412e8f8ebb1fb24e41940fed7f3eb99d7a2d3028/eng/Versions.props#L71-L78

I take it we need to do more here because the PVP is inadvertently changing the value? (And the API version change broke the build or something?)

I would prefer to continue to keep track of this version in Versions.props, and instead add a prefix to the property name to avoid the PVP. I'll push a quick commit to show what I mean.

(Ideally this isn't actually required--in that case, would be nice to not change this at all. 😄)

Reply:
I like the way you're doing it here, I'm going to use this in some other repos too.  This was at least required in the 3.0.1 preview I merged today, if you think it should work now I can give it a try.

Reply:
Yeah, I think the PR as it is now should work the same as the earlier patch to deep hard-code the versions. Validating it would be great if it's not too much trouble. 🙂

Reply:
Yup, this is working locally for me.  Thanks for the improvements!